laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -288,178 +285,92 @@ public function getChoice() {
-        $date = $schedule_data1[0]['date'];
-        $data1['schedule'] = $schedule_data1[0];
-        $data1['attending'] = $attending1;
-
-        //Initiate second choice data
-        $data2 = array(
-          'schedule' => null,
-          'attending' => null
-        );
-        $choice++;
-		// If the second choice exists, get information for second choice
-    	if ($split[1] != 0){
-    		$schedule_data2 = ScheduleData::where('id', $split[1])->get();
-    		$attending_string = $schedule_data2[0]['lead_surgeon'];
-    		$attending = substr($attending_string, strpos($attending_string, "[")+1, strpos($attending_string, "]")-(strpos($attending_string, "[")+1));
-    		$attending2 = substr($attending_string, 0, strpos($attending_string, "["));
-            // Store second choice data
-            $data2['schedule'] = $schedule_data2[0];
-            $data2['attending'] = $attending2;
-	    }
-
-        //Initiate second choice data
-        $data3 = array(
-            'schedule' => null,
-            'attending' => null
-        );
-        $choice++;
-		// If the third choice exists, get information for third choice
-    	if ($split[2] != 0){
-        	$schedule_data3 = ScheduleData::where('id', $split[2])->get();
-    		$attending_string = $schedule_data3[0]['lead_surgeon'];
-    		$attending = substr($attending_string, strpos($attending_string, "[")+1, strpos($attending_string, "]")-(strpos($attending_string, "[")+1));
-    		$attending3 = substr($attending_string, 0, strpos($attending_string, "["));
-            // Store second choice data
-            $data3['schedule'] = $schedule_data3[0];
-            $data3['attending'] = $attending3;
+        for ($i = 0; $i < 3; $i++) {
+            if ($split[$i] != 0) {
+                $data[$i] = array(
+                    'schedule' => null,
+                    'attending' => null,
+                );
+                $choice = $i + 1;
+                $schedule_data[$i] = ScheduleData::where('id', $split[$i])->get();
+                $attending_string = $schedule_data[$i][0]['lead_surgeon'];
+                $attending = substr($attending_string, strpos($attending_string, "[") + 1, strpos($attending_string, "]") - (strpos($attending_string, "[") + 1));
+                $attending1 = substr($attending_string, 0, strpos($attending_string, "["));
+                $date = $schedule_data[$i][0]['date'];
+                $data[$i]['schedule'] = $schedule_data[$i][0];
+                $data[$i]['attending'] = $attending1;
+            }
         }
-        
+
+
         $milestones = Milestone::where('exists', 1)->get();
 
         $anesthesiologists = Anesthesiologist::where('updated_at', '>', Carbon::today())
             ->orderBy('last_name')
             ->get();
-            
-        $datas = compact('data1', 'data2', 'data3');
+
+        $datas = $data;
 
         return view('schedules.resident.milestone', compact('id', 'milestones', 'datas', 'anesthesiologists'));
```

Comment:
You will also need to update the blade as well to use `data` instead of `datas`
We might want to use a better variable name than `data` though. 
Perhaps change it to something else that is more clear like `roomChoices`, `choices`, `scheduleChoices`, or `schedulePreferences`.